internal/source/kubernetes/source.go

```diff
@@ -85,54 +118,82 @@ func (*Source) Stream(ctx context.Context, input source.StreamInput) (source.Str
 		return source.StreamOutput{}, fmt.Errorf("while merging input configs: %w", err)
 	}
 
-	// In Kubernetes, we have an "info" level by default. We should aim to minimize info logging and consider using
-	// the debug level instead. This approach will prevent flooding the Agent logs with irrelevant information,
-	// as the Agent logs everything that plugin writes to stderr.
-	log := loggerx.NewStderr(pkgConfig.Logger{
-		Level: cfg.Log.Level,
+	srcName := input.Context.SourceName
+	eventCh := make(chan source.Event)
+	s.configStore.Store(srcName, SourceConfig{
+		name:                     srcName,
+		eventCh:                  eventCh,
+		cfg:                      cfg,
+		isInteractivitySupported: input.Context.IsInteractivitySupported,
+		clusterName:              input.Context.ClusterName,
+		kubeConfig:               kubeConfig,
 	})
 
-	s := Source{
-		startTime:                time.Now(),
-		eventCh:                  make(chan source.Event),
-		config:                   cfg,
-		logger:                   log,
-		clusterName:              input.Context.ClusterName,
-		kubeConfig:               input.Context.KubeConfig,
-		isInteractivitySupported: input.Context.IsInteractivitySupported,
+	globalSrcCfg, err := s.getGlobalSrcCfg()
+	if err != nil {
+		exitOnError(err, loggerx.New(pkgConfig.Logger{}).WithField("error", err.Error()))
+	}
+
+	id := s.configStore.Len()
+	globalLogger := loggerx.NewStderr(pkgConfig.Logger{
```

Comment:
if that's the "global" one (maybe better `system/internal` (?)) shouldn't we just assign it to `s` instead of passing it as arg? Of course, assign it after `s.bgProcessor.StopAndWait` when we know that all previous configured services are cleaned up.

Reply:
Agree, I've renamed the logger to `system` one 👍 

Yes, as we have mutex on `Stream`, we could assign the logger to `s`.
However, I'd like to keep passing every non-thread-safe argument to the `configureProcessForSources` as value, to avoid unexpected behavior in future if someone modifies the method. Hope that's OK for you 👍